### PR TITLE
Add foreman service to run as puma for DEB

### PR DESCRIPTION
**Pull request by ehelms**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.21
* [ ] 1.20

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: debian/bionic/foreman/control [hunk 1]
```diff
@@ -239,3 +239,14 @@ Description: metapackage providing journald dependencies for Foreman
  a flexible systems management web application.
  .
  The foreman package should be installed along with this package.
+
+Package: foreman-service
+Architecture: all
+Section: web
+Priority: extra
+Depends: foreman (= ${binary:Version}), ruby-puma, libsystemd-dev, ${misc:Depends}
```
Comment (evgeni) [inline debian/bionic/foreman/control]: why does it depend on `libsystemd-dev`?

File: debian/bionic/foreman/control [hunk 2]
```diff
@@ -239,3 +239,14 @@ Description: metapackage providing journald dependencies for Foreman
...
+Description: package providing Foreman services running via puma
+ This package provides Foreman systemd service for Foreman,
+ a flexible systems management web application.
+ .
+ The foreman package should be installed along with this package.
```
Comment (evgeni) [inline debian/bionic/foreman/control]: you depend on it, so this seems rather given?

File: debian/bionic/foreman/control (hunk 1 above)
Comment (ehelms) [reply debian/bionic/foreman/control]: No idea.

File: debian/bionic/foreman/control (hunk 2 above)
Comment (ehelms) [reply debian/bionic/foreman/control]: Just copy pasting and keeping things in line with others.

File: debian/bionic/foreman/control (hunk 1 above)
[the remark this answers is not in the document]
Comment (evgeni) [reply debian/bionic/foreman/control]: There *is* https://packages.debian.org/search?keywords=puma, but we might or might not want to use that.

Comment (evgeni) [reply debian/bionic/foreman/control]: why did you add it? :)

[the remark this answers is not in the document]
Comment (mmoll) [reply debian/bionic/foreman/control]: AFAICT, we don't want to use this, but (like the dynflowd thingies) use the one coming with Foreman's bundled gems.

Comment (ehelms) [reply debian/bionic/foreman/control]: Copy/pastin!

[the remark this answers is not in the document]
Comment (ehelms) [reply debian/bionic/foreman/control]: Soooo, does all that translate to 'remove ruby-puma' here? Or more than that needed?

**Comment by ehelms** [comment]
I made some updates, probably still missing a few things. Treat me like I am completely new at this if more changes are needed. Spell it out for me :)

File: debian/bionic/foreman/control (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmoll) [reply debian/bionic/foreman/control]: Exactly, removing it as dependency. Also, akeen to `dynflowd.service` in DEB packaging, at the end the puma process needs to get started via bundle.

**Comment by evgeni** [comment]
Can you call your branch `deb/develop-foreman-puma` so that jenkins properly fetches the tarball from jenkins and not from downloads?

**Comment by evgeni** [comment]
As the service file is now in the tarball, there is nothing that would install it into the .deb (contrary to the case where it's inside `debian/`, where some magic happen).

You'll need to add a line like this: https://github.com/theforeman/foreman-packaging/blob/deb/develop/debian/stretch/foreman/foreman-test.install with `/usr/lib/systemd/system` as target.

One thing that makes me think: We also ship a traditional `foreman.init` file for SysV init, which will be used by systemd if there is no `foreman.service`. Should we drop that now?

**Comment by ekohl** [comment]
> Should we drop that now?

I think I heard some users mention they use sysvinit only, but I'd be fine with dropping it. Probably good to write up something for the release notes about what the service is and how it's supported though.

**Comment by ehelms** [comment]
Closing in favor of https://github.com/theforeman/foreman-packaging/pull/3617

**Pull request closed without merging**